docs/reference/ingest/processors/data-stream-router.asciidoc

```diff
@@ -0,0 +1,43 @@
+[[data-stream-router-processor]]
+=== Data stream router processor
+++++
+<titleabbrev>Data stream router</titleabbrev>
+++++
+
+The `data_stream_router` processor allows to route a document from one data stream to another data stream.
+It can use both static values or values from the document to determine the target data stream.
+
+The name of a data stream is comprised of three parts and looks like this: `<type>-<dataset>-<namespace>`.
+See the {fleet-guide}/data-streams.html#data-streams-naming-scheme[data stream naming scheme] documentation for more details.
+
+NOTE: `data_stream_router` processor can only be used on data streams that follow the data streams naming scheme.
+Trying to use this processor on a data stream with a non-compliant name will raise an exception.
+
+After a `data_stream_router` processor has been executed, all the other processors of the current pipeline are skipped.
+This means that at most one `data_stream_router` processor is ever executed within a pipeline,
+allowing to define mutually exclusive routing conditions,
+similar to a if, else-if, else-if, … condition.
+
+[[data-stream-router-options]]
+.Data stream router options
+[options="header"]
+|======
+| Name        | Required  | Default  | Description
+| `dataset`   | no        | -        | A static value for the dataset part of the data stream name. In addition to the criteria for <<indices-create-api-path-params, index names>>, cannot contain `-` and must be no longer than 100 characters. Example values are `nginx.access` and `nginx.error`. If not set, gets the value of the field `data_stream.dataset` from the document. When using values from the document, the processor replaces invalid characters with `_`. If the option is not set and the document also doesn't contain a corresponding field, it uses the `<dataset>` part of the index name as a fallback.
+| `namespace` | no        | -        | A static value for the namespace part of the data stream name. See the criteria for <<indices-create-api-path-params, index names>> for allowed characters. Must be no longer than 100 characters. If not set, gets the value of the field `data_stream.namespace` from the document. When using values from the document, the processor replaces invalid characters with `_`. If the option is not set and the document also doesn't contain a corresponding field, it uses the `<namespace>` part of the index name as a fallback.
+include::common-options.asciidoc[]
+|======
+
+NOTE: It's not possible to change the `type` of the data stream by setting the `data_stream.type` in the document.
+
+[source,js]
+--------------------------------------------------
+{
+  "data_stream_router": {
+    "tag": "nginx",
+    "if" : "ctx?.log?.file?.path?.contains('nginx')",
```

Comment:
I remember there were some efforts to simplify this. Do we still need `ctx`? Would be nice if users could just reference a field, no matter where it is stored.

Same for all the `?.`, users should not have to care and just type it in. I know, more an underlying ES issue but maybe we already work around it in the code of this processor.